src/client/views/cube-view/downloadable-dataset-context.tsx

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2017-2021 Allegro.pl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Dataset, TabulatorOptions } from "plywood";
+import * as React from "react";
+import { ReactNode } from "react";
+import { Nullary, Unary } from "../../../common/utils/functional/functional";
+
+export interface DataSetWithTabOptions {
+  dataset: Dataset;
+  options: TabulatorOptions;
+}
+
+export interface DownloadableDataset {
+  getDataset: Nullary<DataSetWithTabOptions | null>;
+  setDataset: Unary<DataSetWithTabOptions, void>;
+}
+
+export const DownloadableDatasetContext = React.createContext<DownloadableDataset>({
+  getDataset(): DataSetWithTabOptions {
+    throw new Error("Trying to fetch DownloadableDataset before Provider initialisation");
+  },
+  setDataset(dataset: DataSetWithTabOptions): void {
+    throw new Error("Trying to set DownloadableDataset before Provider initialisation");
+  }
+});
+
+interface DownloadableDatasetProps {
+  children: ReactNode;
+}
+
+export class DownloadableDatasetProvider extends React.Component<DownloadableDatasetProps> {
+
+  private dataset?: DataSetWithTabOptions = null;
+  private value: DownloadableDataset = {
```

Comment:
It is really a React 16.x `useRef`. In context we share stable object with getter and setter. We don't want to serenader anything because this dataset changes. It is only lazily accessed reference.